src/main/java/org/folio/rs/service/RetrievalQueueService.java

```diff
@@ -96,4 +138,52 @@ private void saveRetrievalQueueWithCurrentDate(RetrievalQueueRecord record) {
     return (record, criteria, builder) -> builder.equal(record.get(ID), stringToUUIDSafe(id));
   }
 
+  private LocationMapping getLocationMapping(Item item) {
+    return Objects.nonNull(item.getEffectiveLocation())
+        ? locationMappingsService.getMappingByFolioLocationId(item.getEffectiveLocation().getId())
+        : null;
+  }
+
+  private Item getOriginalItemByBarcode(MovedEventRequest movedEventRequest) {
+    ResultList<Item> items = inventoryClient.getItem("barcode==" + movedEventRequest.getItemBarCode());
+    if (Objects.isNull(items)) {
+      throw new EntityNotFoundException("Item with barcode " + movedEventRequest.getItemBarCode() + NOT_FOUND);
+    }
+    return items.getResult().get(0);
+  }
+
+  private User getUserByRequesterId(MovedEventRequest movedEventRequest) {
+    ResultList<User> users = usersClient.query("id==" + movedEventRequest.getRequesterId());
+    if (Objects.isNull(users)) {
+      throw new EntityNotFoundException("User with id " + movedEventRequest.getRequesterId() + NOT_FOUND);
+    }
+    return users.getResult().get(0);
+  }
+
+  private RetrievalQueueRecord buildRetrievalQueueRecord(MovedEventRequest movedEventRequest,
+      Item item, User patron, LocationMapping mapping, FolioLocation folioLocation) {
+    return RetrievalQueueRecord.builder()
+        .id(UUID.randomUUID())
+        .holdId(movedEventRequest.getHoldId())
+        .patronBarcode(patron.getBarcode())
+        .patronName(patron.getUsername())
+        .callNumber(getCallNumber(item))
+        .itemBarcode(movedEventRequest.getItemBarCode())
+        .createdDateTime(LocalDateTime.now())
+        .pickupLocation(folioLocation.getCode())
+        .requestStatus(movedEventRequest.getRequestStatus())
+        .requestNote(movedEventRequest.getRequestNote())
+        .remoteStorageId(stringToUUIDSafe(mapping.getConfigurationId()))
+        .instanceTitle(item.getTitle())
+        .instanceAuthor(item.getContributorNames()
```

Comment:
Here could be null pointer exception if item.getContributors()  will return null